examples/decoding/plot_decoding_ssd_csp_eeg.py

```diff
@@ -0,0 +1,256 @@
+raw.set_montage(montage)
+
+# strip channel names of "." characters
+raw.rename_channels(lambda x: x.strip('.'))
+# for the sake of comparison, we save the raw object before filteing
+raw_ssd = raw.copy()
+##############################################################################
+# Data filtering.
+
+# we are going to filter data in the alpha band
+freq_signal = [8, 12]
+# apply band-pass filter
+raw.filter(freq_signal[0], freq_signal[1], fir_design='firwin',
+           skip_by_annotation='edge')
+
+events, _ = events_from_annotations(raw, event_id=dict(T1=2, T2=3))
+
+picks = pick_types(raw.info, meg=False, eeg=True, stim=False, eog=False,
+                   exclude='bads')
+
+# Extract epochs between 1 and 2s
+epochs = Epochs(raw, events, event_id, tmin, tmax, proj=True, picks=picks,
+                baseline=None, preload=True)
+epochs_cropped = epochs.copy().crop(tmin=1., tmax=2.)
+labels = epochs.events[:, -1] - 2
+epochs_data = epochs_cropped.get_data()
+###############################################################################
+# Traditional CSP+LDA pipeline
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+# define pipelines with monte-carlo simulations
+cv = ShuffleSplit(5, test_size=0.2, random_state=42)
+# pipeline methods
+lda = LinearDiscriminantAnalysis()
+csp = CSP(n_components=4, reg=None, log=True, norm_trace=False)
+pipe_csp = Pipeline([('CSP', csp), ('LDA', lda)])
+# this is the accuracy we would like to improve by adding SSD into the pipeline
+scores_csp = cross_val_score(pipe_csp, epochs_data, labels, cv=cv,
+                             n_jobs=1).mean()
+###############################################################################
+# SSD enhances SNR, thus what is considered 'noise' should be defined
+# Typically the 2 Hz sourronunding frequencies are taken
+freq_noise = [6, 14]
+# when applying SSD, data should be broader band filtered
+# a broader band of the freq_signal is defined
+freq_ssd = [5, 15]
+# as before, filter data
+raw_ssd.filter(freq_ssd[0], freq_ssd[1], fir_design='firwin',
+               skip_by_annotation='edge')
+###############################################################################
+# SSD + CSP + LDA pipeline
+# ^^^^^^^^^^^^^^^^^^^^^^^^^
+
+# SSD can be applied either BEFORE or AFTER data epoching, each approach has
+# it owns advantages and disadvantages. Here both approaches are going to be
+# implemented.
+###############################################################################
+# SSD outside the pipeline
+# ^^^^^^^^^^^^^^^^^^^^^^^^
+
+# copy raw data
+raw_ssd_transformed = raw_ssd.copy()
+# define SSD filter parameters
+filt_params_signal = dict(l_freq=freq_signal[0], h_freq=freq_signal[1],
+                          l_trans_bandwidth=4, h_trans_bandwidth=4)
+filt_params_noise = dict(l_freq=freq_noise[0], h_freq=freq_noise[1],
+                         l_trans_bandwidth=4, h_trans_bandwidth=4)
+
+# Since we are working with a high electrode counting dataset (64 channels),
```

Comment:
```suggestion
# Since we are working with a data set with many electrodes (64 channels),
```

Reply:
(Although it probably depends highly on the context whether 64 electrodes is considered many ... ;-)